File: src/CorporateMessengerWithAgile/Application/Dto/WorkplaceDto.cs
namespace Application.Dto
{
    public class WorkplaceDto : BaseDto
    {
        public CompanyDto Company { get; set; } = null!;
        public PositionInCompanyDto PositionInCompany { get; set; } = null!;
        public UserDto User { get; set; } = null!;
        public IReadOnlyList<TeamMemberDto> TeamMembers { get; set; } = [];
    }
}

File: src/CorporateMessengerWithAgile/ReactApp/reactapp.client/src/models/entity/WorkplaceDto.ts
// src/models/entity/WorkplaceDto.ts
import { z } from 'zod';
import { BaseDtoSchema } from './BaseDto';
import { CompanyDtoSchema } from './CompanyDto';
import { PositionInCompanyDtoSchema } from './PositionInCompanyDto';
import { TeamMemberDtoSchema } from './TeamMemberDto';

export const WorkplaceSchema = BaseDtoSchema.extend({
    company: CompanyDtoSchema,
    positionInCompany: PositionInCompanyDtoSchema,
    teamMembers: z.array(TeamMemberDtoSchema).optional(),
});

export type WorkplaceDto = z.infer<typeof WorkplaceSchema>;
